### PR TITLE
Drop jprocesses in favor of java.lang.ProcessHandle

### LINKED ISSUE #16521: Drop jprocesses in favor of java.lang.ProcessHandle
## Description
As per https://github.com/quarkusio/quarkus/pull/16249#issuecomment-815263204

[`jprocesses`](https://github.com/profesorfalken/jProcesses) has not seen any activity for 3 years and with JDK 8 support now dropped (in main), the OOTB `ProcessHandle` should be used instead.

## Implementation ideas
Remove code, write a tiny bit of new code, remove dependency - be happy! 😉

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #16521

### CODE REVIEW

File: test-framework/devmode-test-utils/src/main/java/io/quarkus/test/devmode/util/DevModeTestUtils.java [hunk 1]
```diff
@@ -16,20 +16,24 @@
...
-            for (String part : cmdParts) {
-                if (pi.getCommand().contains(part)) {
-                    JProcesses.killProcess(Integer.parseInt(pi.getPid()));
-                    break;
-                }
-            }
-        }
+        ProcessHandle.allProcesses()
+                .filter(processHandle -> {
+                    String cmdLine = processHandle.info().commandLine().orElse(null);
```
Comment (famod) [inline test-framework/devmode-test-utils/src/main/java/io/quarkus/test/devmode/util/DevModeTestUtils.java:25]: Just a personal thing I guess, but I would do the filtering like this (might contain a quirk, haven't compiled):
```
                .filter(processHandle -> processHandle.info().commandLine()
                    .map(line -> cmdParts.stream().anyMatch(line::contains))
                    .orElse(false))
```

**Review by famod — APPROVED** [review]
PS: I think the old as well as the new solution have the same nasty trait of killing unrelated process instances when working locally (e.g. you have quarkus:dev running in some other project and run a Quarkus test in this repo).
But this is nothing to block this PR.

File: test-framework/devmode-test-utils/src/main/java/io/quarkus/test/devmode/util/DevModeTestUtils.java (hunk 1 above)
Comment (gastaldi) [reply test-framework/devmode-test-utils/src/main/java/io/quarkus/test/devmode/util/DevModeTestUtils.java]: How about:
```java
    public static void killProcesses(String... cmdParts) {
        Stream<String> cmdPartsStream = Arrays.stream(cmdParts);
        ProcessHandle.allProcesses()
                .filter(processHandle -> processHandle.info().commandLine().stream()
                        .anyMatch(line -> cmdPartsStream.anyMatch(line::contains)))
                .forEach(ProcessHandle::destroyForcibly);
    }
``` ?

Comment (famod) [reply test-framework/devmode-test-utils/src/main/java/io/quarkus/test/devmode/util/DevModeTestUtils.java]: Perfection!

Comment (gastaldi) [reply test-framework/devmode-test-utils/src/main/java/io/quarkus/test/devmode/util/DevModeTestUtils.java]: image

**Comment by gsmet** [comment]
Looks like CI is unhappy. Thanks for testing the bot :).

**Comment by famod** [comment]
Gradle tests should be better with the memory tweaks I did in #16249.

Maven test job seems to show that killing does not work properly?
> ERROR [io.qua.run.Application] (Quarkus Main Thread) Port 8080 seems to be in use by another process. Quarkus may already be running or the port is used by another application.

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
The test failures are very suspicious. These tests weren't flaky.

**Comment by famod** [comment]
Yes, looks very related, especially that port error log I commented on earlier.

In the worst case, `ProcessHandle.allProcesses()` sees fewer processes than `jprocesses`. This needs a deep dive.

**Comment by famod** [comment]
FYI, I'm having a look at this in my fork. It's remarkable that only Windows jobs are affected.
I'll remove the watiting-for-ci label for the time being.

**Comment by famod** [comment]
This doesn't look good: [JDK-8176725 ProcessHandle arguments is always null](https://bugs.openjdk.java.net/browse/JDK-8176725?focusedCommentId=14061174&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14061174)

> The command line arguments are not available via a non-privileged API for other processes
and so the Optional is always empty. The API is explicit that the values are OS specific. 
> If in the future, the arguments are available by a Window APIs, the implementation can be updated.

Investigating workarounds...

**Comment by famod** [comment]
Ok, I think I found a good workaround. Will see how CI fares in my fork. If all goes to plan I'll create a fresh PR and close this one...

**Comment by gastaldi** [comment]
@famod you can also push to my branch if you want: 

`git push git@github.com/gastaldi/quarkus.git remove_jprocessescd -f`

**Review by famod — COMMENTED** [review]
Ok, pushed to your branch @gastaldi

File: test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/RunningInvoker.java [hunk 2]
```diff
@@ -88,11 +88,12 @@ public void stop() {
         if (result == null) {
             return;
         }
+        // Kill all processes that were (indirectly) spawned by the current process.
+        // It's important to do it this way (instead of calling result.destroy() first)
+        // because otherwise children of that process can become orphaned zombies.
+        DevModeTestUtils.killDescendingProcesses();
```
Comment (famod) [inline test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/RunningInvoker.java:94]: This invocation order problem took the most time to figure out. I was able to see (on Windows, via Process Explorer) that sometimes a sub-branch of processes survives `result.destroy()`, loosing that parent and thus also the transitive parent connection to the surefire test JVM process. That resulted in `descendants()` not finding those "orphaned" processes anymore.

**Comment by famod** [comment]
All green! @geoand @gastaldi @gsmet (tripple g 😉) do you want to have another look?

**Review by geoand — APPROVED** [review]

**Pull request merged**